contracts/tips/src/state.rs

```diff
@@ -0,0 +1,311 @@
+use crate::error::ContractError;
+use crate::msg::ServiceFee;
+use crate::utils::sum_coins_sorted;
+use cosmwasm_std::{Addr, Coin, Decimal, Uint128};
+use cw_storage_plus::{Item, Map};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::VecDeque;
+use std::convert::TryFrom;
+use std::ops::{Div, Mul};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum StateServiceFee {
+    Fixed { amount: Vec<Coin> },
+    Percentage { value: Decimal },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Config {
+    pub admin: Addr,
+    pub subspace_id: u64,
+    pub service_fee: Option<StateServiceFee>,
+    pub tips_history_size: u32,
+}
+
+pub type SentTip = (Addr, Vec<Coin>, u64);
+pub type ReceivedTip = (Addr, Vec<Coin>, u64);
+pub type PostTip = (Addr, Vec<Coin>);
+
+pub const CONFIG: Item<Config> = Item::new("config");
+pub const SENT_TIPS_HISTORY: Map<Addr, VecDeque<SentTip>> = Map::new("sent_tips_history");
+pub const RECEIVED_TIPS_HISTORY: Map<Addr, VecDeque<ReceivedTip>> =
+    Map::new("received_tips_history");
+pub const POST_TIPS_HISTORY: Map<u64, VecDeque<PostTip>> = Map::new("received_tips_history");
```

Comment:
```suggestion
pub const POST_TIPS_HISTORY: Map<u64, VecDeque<PostTip>> = Map::new("post_tips_history");
```

Reply:
Thanks this was a very bad copy paste mistake :sweat_smile: